File: Calculadora_v2.php
<?php

$operacion = readline("Ingrese operación:");
$limpia1 = preg_replace('/\s+/', ' ', $operacion);
$limpia2 = trim($limpia1);
$operadores = explode(' ', $limpia2);
// print_r($operadores);

$num1 = intval($operadores[0]);
$num2 = intval($operadores[2]);
$operador = $operadores[1];

function suma($num1, $num2){
    $resultado = $num1 + $num2;
    return $resultado;
}

function resta($num1, $num2){
    $resultado = $num1 - $num2;
    return $resultado;
}

function division($num1, $num2){
    if($num2 == 0){
        return "No se puede dividir entre cero";
    }
    $resultado = $num1 / $num2;
    return $resultado;
}

function multiplicacion($num1, $num2){
    $resultado = $num1 * $num2;
    return $resultado;
}

if($operador == "+"){

    echo suma($num1,$num2);

}elseif($operador == "-"){

    echo resta($num1,$num2);

}elseif($operador == "/"){

    echo division($num1,$num2);
    
}elseif($operador == "*"){

    echo multiplicacion($num1,$num2);
    
}else{

    echo"Esta operacion  es incorrecto";
}

?>

File: Calculadora_v2 .py
import re

operacion = input("Ingrese operación:")
limpia1 = re.sub('/\s+/', ' ', operacion)
limpia2 = limpia1.strip()
operadores = limpia2.split(' ')

num1 = int(operadores[0])
num2 = int(operadores[2])
operador = operadores[1]

def suma(num1, num2):
    resultado = num1 + num2
    return resultado


def resta(num1, num2):
    resultado = num1 - num2
    return resultado


def division(num1, num2):
    if num2 == 0 :
        return "No se puede dividir entre cero"
    resultado = num1 / num2
    return resultado


def multiplicacion(num1, num2):
    resultado = num1 * num2
    return resultado


if operador == "+":
    print(suma(num1, num2))

elif operador == "-":
    print(resta(num1, num2))

elif operador == "/":
    print(division(num1, num2)) 
    
elif operador == "*":
    print(multiplicacion(num1, num2)) 
    
else:
    print("Esta operacion  es incorrecto")


